src/Interpreters/Context.cpp

```diff
@@ -1816,6 +1816,57 @@ std::optional<UInt16> Context::getTCPPortSecure() const
     return {};
 }
 
+// Small compile time FNV-1a 64 hashing.
+constexpr size_t compileTimeHashing(const std::string_view & s)
+{
+    size_t res = 0xcbf29ce484222325;
+    for (auto c : s)
+    {
+        res ^= c;
+        res *= 1099511628211;
+    }
+    return res;
+}
+
+constexpr size_t operator"" _h(const char * p, size_t s)
+{
+    return compileTimeHashing(std::string_view(p, s));
+}
+
+UInt16 Context::getPort(const std::string_view & port_name) const
+{
+    auto h = compileTimeHashing(port_name);
+    auto lock = getLock();
+    const auto & config = getConfigRef();
+    switch (h)
+    {
+        case "tcp"_h:
+            return config.getInt("tcp_port", DBMS_DEFAULT_PORT);
```

Comment:
It is technically incorrect to return default port if it is not defined in config.
Because in this case the port is not used by server.

Reply:
Ah, ok.